lib/etcdv3/auth.rb

```diff
@@ -47,74 +49,80 @@ def role_grant_permission(name, permission, key, range_end)
     end
 
-    def role_list
+    def role_list(timeout: nil)
       request = Etcdserverpb::AuthRoleListRequest.new
-      @stub.role_list(request, metadata: @metadata)
+      @stub.role_list(request, metadata: @metadata, deadline: Time.now.to_f  + (timeout || @timeout))
     end
 
-    def user_list
+    def user_list(timeout: nil)
       request = Etcdserverpb::AuthUserListRequest.new
-      @stub.user_list(request, metadata: @metadata)
+      @stub.user_list(request, metadata: @metadata, deadline: Time.now.to_f  + (timeout || @timeout))
     end
 
-    def user_add(user, password)
+    def user_add(user, password, timoeut: nil)
       request = Etcdserverpb::AuthUserAddRequest.new(
         name: user,
         password: password
       )
-      @stub.user_add(request, metadata: @metadata)
+      @stub.user_add(request, metadata: @metadata, deadline: Time.now.to_f  + (timeout || @timeout))
     end
 
-    def user_delete(user)
+    def user_delete(user, timeout: nil)
       request = Etcdserverpb::AuthUserDeleteRequest.new(name: user)
-      @stub.user_delete(request, metadata: @metadata)
+      @stub.user_delete(request, metadata: @metadata, deadline: Time.now.to_f  + (timeout || @timeout))
     end
 
-    def user_get(user)
+    def user_get(user, timeout: nil)
       request = Etcdserverpb::AuthUserGetRequest.new(name: user)
-      @stub.user_get(request, metadata: @metadata)
+      @stub.user_get(request, metadata: @metadata, deadline: Time.now.to_f  + (timeout || @timeout))
     end
 
-    def user_change_password(user, new_password)
+    def user_change_password(user, new_password, timeout: nil)
       request = Etcdserverpb::AuthUserChangePasswordRequest.new(
         name: user,
         password: new_password
       )
-      @stub.user_change_password(request, metadata: @metadata)
+      @stub.user_change_password(request, metadata: @metadata, deadline: Time.now.to_f  + (timeout || @timeout))
     end
 
-    def user_grant_role(user, role)
+    def user_grant_role(user, role, timeout: nil)
       request = Etcdserverpb::AuthUserGrantRoleRequest.new(user: user, role: role)
-      @stub.user_grant_role(request, metadata: @metadata)
+      @stub.user_grant_role(request, metadata: @metadata, deadline: Time.now.to_f  + (timeout || @timeout))
     end
 
-    def user_revoke_role(user, role)
+    def user_revoke_role(user, role, timeout: nil)
       request = Etcdserverpb::AuthUserRevokeRoleRequest.new(name: user, role: role)
-      @stub.user_revoke_role(request, metadata: @metadata)
+      @stub.user_revoke_role(request, metadata: @metadata, deadline: Time.now.to_f  + (timeout || @timeout))
     end
 
-    def generate_token(user, password)
+    def generate_token(user, password, timeout: nil)
       request = Etcdserverpb::AuthenticateRequest.new(
         name: user,
         password: password
       )
-      @stub.authenticate(request).token
+      @stub.authenticate(request, deadline: Time.now.to_f  + (timeout || @timeout)).token
```

Comment:
Looks like you defined a private method `deadline`.  We should probably just use that eh?